File: rb/features.rb
# Euronews SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module EuronewsFeatures
  def self.make_feature(name)
    case name
    when "base"
      EuronewsBaseFeature.new
    when "ratelimit"
      EuronewsRatelimitFeature.new
    when "retry"
      EuronewsRetryFeature.new
    when "test"
      EuronewsTestFeature.new
    when "timeout"
      EuronewsTimeoutFeature.new
    else
      EuronewsBaseFeature.new
    end
  end
end

File: py/features.py
# Euronews SDK feature factory

from feature.base_feature import EuronewsBaseFeature
from feature.test_feature import EuronewsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EuronewsBaseFeature(),
        "test": lambda: EuronewsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
